File: thunder_gui/src/utils/package.ts
import { Card } from "./card";

export interface PackageList {
    bg_image_src?: string,
    categories: never[],
    community_name: string,
    has_more_pages: boolean,
    packages: Card[],
}

export interface PackageVersionExperimental {
    namespace?: string,
    name: string,
    description: string,
}

export interface ThunderstorePackage {
    namespace?: string,
    name: string,
    full_name: string,
    owner?: string,
    package_url?: string,
    date_created?: string,
    date_updated?: string,
    rating_score?: string,
    is_pinned?: boolean,
    is_deprecated?: boolean,
    total_downloads?: string,
    latest: PackageVersionExperimental,
    community_listings?: never[]
}

export interface ThunderstorePackageV1 {
    name?: string,
    full_name?: string,
    owner?: string,
    package_url?: string,
    donation_link?: string,
    date_created?: string,
    date_updated?: string,
    uuid4?: string,
    is_pinned?: string,
    is_deprecated?: string,
    has_nsfw_content?: boolean,
    categories?: string,
    versions?: string,
}

export type PackageListV1 = ThunderstorePackageV1[];

File: thunder_manager_common/src/model/thunderstore/package.rs
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
#[serde(untagged)]
pub enum StringOri32 {
    String(String),
    I32(i32)
}

pub type RatingScore = StringOri32;
pub type DownloadCount = StringOri32;

#[derive(Deserialize, Serialize)]
pub struct PackageVersionExperimental {
    pub namespace: Option<String>,
    pub name: String,
    pub version_number: String,
    pub full_name: Option<String>,
    pub description: String,
}

#[derive(Deserialize, Serialize)]
pub struct ExperimentalPackage {
    pub namespace: Option<String>,
    pub name: String,
    pub full_name: Option<String>,
    pub owner: Option<String>,
    pub package_url: Option<String>,
    pub date_created: Option<String>,
    pub date_updated: Option<String>,
    pub rating_score: Option<RatingScore>,
    pub is_pinned: Option<bool>,
    pub is_deprecated: Option<bool>,
    pub total_downloads: Option<DownloadCount>,
    pub latest: PackageVersionExperimental,
}